build/pack.c

```diff
@@ -726,16 +727,37 @@ static rpmRC packageBinary(rpmSpec spec, Package pkg, const char *cookie, int ch
     return rc;
 }
 
+static int compareBinaries(const void *p1, const void *p2) {
+    Package pkg1 = *(Package *)p1;
+    Package pkg2 = *(Package *)p2;
+    uint64_t size1 = headerGetNumber(pkg1->header, RPMTAG_SIZE);
+    uint64_t size2 = headerGetNumber(pkg2->header, RPMTAG_SIZE);
+    return size1 - size2;
```

Comment:
The difference between two 64bit integers can overflow the 32bit int return code, you'll want to normalize the return one way or the other.